tests/attendance.test.ts

```diff
@@ -0,0 +1,223 @@
+import * as faker from 'faker';
+import { ActivityType, UserAccessType } from '../types';
+import { ControllerFactory } from './controllers';
+import { DatabaseConnection, EventFactory, PortalState, UserFactory } from './data';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('attendance', () => {
+  test('only admins can see attendances for event', async () => {
+    const conn = await DatabaseConnection.get();
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+    const staff = UserFactory.fake({ accessType: UserAccessType.STAFF });
+    const member = UserFactory.fake();
+    const event = EventFactory.fake({ requiresStaff: true });
+
+    await new PortalState()
+      .createUsers(admin, staff, member)
+      .createEvents(event)
+      .attendEvents([staff, member], [event], true)
+      .write();
+
+    const attendanceController = ControllerFactory.attendance(conn);
+    const params = { uuid: event.uuid };
+
+    // throws permissions error for member
+    await expect(attendanceController.getAttendancesForEvent(params, member))
+      .rejects.toThrow();
+
```

Comment:
nit: we should probably check that specifically a `ForbiddenError` was thrown vs. any error, i.e.
```ts
await expect(attendanceController.getAttendancesForEvent(params, member))
      .rejects.toThrow(ForbiddenError);
```

Reply:
Ah yeah definitely, updated.